app/controllers/root_controller.rb

```diff
@@ -109,12 +110,58 @@ def publication
       end
 
       @interaction_details = prepare_interaction_details(@publication, authority_slug, snac)
+    elsif @publication.format == 'local_transaction'
+      @location = fetch_location @postcode
+
+      if @location.nil?
+        @postcode_error = "fullPostcodeNoMapitMatch"
```

Comment:
Oh. I see that it might also return `nil` when mapit can't find a match.  Maybe we could refactor `fetch_location` to return something more useful than a `nil`, a `Location`, or an `GdsApi::BaseError` subclass.